library(QTLseqr)
library("ggplot2")


HighBulk <- "H1706Z" ## é«˜å€¼æ··æ± åç§?
LowBulk <- "H1706F"  ## ä½Žå€¼æ··æ± åç§?
Chroms <-paste0("arahy.Tifrunner.gnm2.Arahy.",c("01","02","03","04","05","06","07","08","09","10",
                                                "11","12","13","14","15","16","17","18","19","20"))
## SNPæ•°æ®è¯»å–
df <- importFromGATK(file = choose.files() ,
    highBulk = HighBulk, # æŒ‡å®šé«˜å€¼æ··æ±?
    lowBulk = LowBulk, # æŒ‡å®šä½Žè‡³æ··æ± 
    chromList = Chroms # æŒ‡å®šåˆ†æžç”¨æŸ“è‰²ä½“åˆ—è¡¨
    ) 


## ç»˜åˆ¶æ·±åº¦åˆ†å¸ƒå›? 

p1 <- ggplot(data = df) +
  geom_histogram(aes(x = DP.HIGH + DP.LOW)) +
  xlim(0,800)
  
pdf(file="SNP_depth.pdf")
p1
dev.off()

## ç»˜åˆ¶refç­‰ä½åŸºå› é¢‘çŽ‡åˆ†å¸ƒå›?
p2 <- ggplot(data = df) +
  geom_histogram(aes(x = REF_FRQ))
  
pdf(file="ref_allele_frequency.pdf")
p2
dev.off()


## ç»˜åˆ¶é«˜å€¼æ··æ± SNP-indexåˆ†å¸ƒ
p3 <- ggplot(data = df) +
  geom_histogram(aes(x = SNPindex.HIGH))
pdf(file="SNPindex.HIGH.dis.pdf")
p3
dev.off()

## ç»˜åˆ¶ä½Žå€¼æ··æ± SNP-indexåˆ†å¸ƒ
p4 <- ggplot(data = df) +
  geom_histogram(aes(x = SNPindex.LOW))

pdf(file="SNPindex.LOW.dis.pdf")
p4
dev.off()

## SNPè¿‡æ»¤
df <- subset(df, !is.na(SNPindex.LOW) & !is.na(SNPindex.HIGH))
df_filt <- filterSNPs(
    SNPset = df,
    refAlleleFreq = 0.20, ## ref alleleé¢‘çŽ‡è¿‡æ»¤ï¼?0.2è¡¨ç¤º0.2~0.8ä¹‹é—´
    minTotalDepth = 100, ## æœ€å°æ·±åº¦è¿‡æ»?
    maxTotalDepth = 400, ## æœ€å¤§æ·±åº¦è¿‡æ»?
    minSampleDepth = 5, ## å•ä¸ªæ ·å“æœ€å°æ·±åº?
    minGQ = 99, ## genotype quality è¿‡æ»¤
    verbose = TRUE  ## è¾“å‡ºæ—¥å¿—
  )

## è¿›è¡ŒdeltaSNPindex è®¡ç®—
df_filt <- runQTLseqAnalysis(df_filt,
    windowSize = 3e6, ## çª—å£å¤§å°
    popStruc = "F2",  ## ç¾¤ä½“ç±»åž‹ï¼ŒF2 æˆ–è€…RIL
    bulkSize = c(25,25), ## æ··æ ·ä¸ªæ•°ï¼Œç¬¬ä¸€ä¸ªé«˜å€¼ç»„
    replications = 10000, ## bootstrapæ¬¡æ•°
    intervals = c(95, 99) ## ç½®ä¿¡åŒºé—´
)


## SNP æ²¿æŸ“è‰²ä½“åˆ†å¸ƒå›?
p5 <- plotQTLStats(SNPset = df_filt, var = "nSNPs")
pdf(file="SNP_filter.window.pdf", width = 20, height = 4)
p5
dev.off()

## deltaSNPindexæ²¿æŸ“è‰²ä½“åˆ†å¸ƒ
p6 <- plotQTLStats(SNPset = df_filt, var = "deltaSNP", plotIntervals = TRUE)
pdf(file="deltaSNPindex.pdf", width = 20, height = 4)
p6
dev.off()

## æå–æ˜¾è‘—æ€§åŒºåŸ?
QTL <- getSigRegions(SNPset = df_filt, 
    method = "QTLseq", interval = 95)

## è¾“å‡ºåˆ°æ–‡ä»?
write.table(QTL[[1]],
    "QTLseq_result.SigRegions.table",
    sep="\t",quote=F)

results95 <- getQTLTable(
  SNPset = df_filt,  
  method = "QTLseq", 
  interval = 95,  ## ç»“æžœé˜ˆå€?
  export = FALSE)

## è¾“å‡ºåˆ°æ–‡ä»?
write.table(results95,
            file="QTLseq_result_deltaSNPindex_CI95.table" ,
            sep="\t", quote = F)


## æå–QTLseqç»“æžœ
results99 <- getQTLTable(
  SNPset = df_filt,  
  method = "QTLseq", 
  interval = 99,  ## ç»“æžœé˜ˆå€?
  export = FALSE)

## è¾“å‡ºåˆ°æ–‡ä»?
write.table(results99,
    file="QTLseq_result_deltaSNPindex_CI99.table" ,
    sep="\t", quote = F)

